### ISSUE TITLE
Priorizing the typeof Entity within a Colshape

### ISSUE THREAD

**Issue by AlteredCarbonate** [altmp/altv-issues#560]
Priorizing what typeof Entity is entering / leaving the Colshape, not based on what hits first.

**Is your feature request related to a problem? Please describe.**
I had to rework my code as I found out about this problem, to match both possiblites.
If it's a vehicle or player.

**Describe the solution you'd like**
The point isn't that the event only proceeds if it is typeof alt.vehicle, but that it forces to get this type.
As it depends right now, on what hits the colshape first or what the server detects first.
Yes there's an workaround (Seen on alternatives you've considered), but maybe there's a way?

Here's an example:
A Player who's driving a car enters the mentioned Colshape.

If we now, want to get the vehicle we could proceed like this:
```javascript
alt.on('entityEnterColshape', (colshape: Colshape, entity: Entity, typeof: alt.vehicle) => {
    if (colshape == 'wonderfulColshape') {
        // Priorized for Vehicle
    }
});
```
And if we've choosen to get the Player, within the car:
```javascript
alt.on('entityEnterColshape', (colshape: Colshape, entity: Entity, typeof: alt.player) => {
    if (colshape == 'wonderfulColshape') {
        // Priorized for Player
    }
});
```

**Describe alternatives you've considered**
This is the current approach, to create the code for both possibilites.
As it is a 50 : 50 Chance what you get as entity - Double the work
```javascript
alt.on('entityEnterColshape', (colshape, entity) => {
    if (colshape == 'wonderfulColshape') {
        if (typeof entity === alt.vehicle) {
            // DO STUFF HERE
        }
        if (typeof entity === alt.player) {
            // DO STUFF HERE
        }
    }
});
```

**Additional context**

A lot of people would just suggest right now =>

> Why don't you just accept it and work with it, as you do right now?

Because it might help out others and make things easier!
Alright, that's it from me - Take care & stay healthy, Carbon 😷

**Comment by FabianTerhorst** [comment]
I feel like when this influences the performance of the colshape thread its not needed.

**Comment by robertnisipeanu** [comment]
Why not register the event two times if you want to do that in two different function and not in an if/else if. You can even even reverse the if parameters and return inside, so you get away from nested ifs.

Something similar:
```
alt.on('entityEnterColshape', (colshape: Colshape, entity: Entity) => {
    if(!entity instanceof alt.Player) return;
    // Entity is instance of alt.Player

    if(colshape != 'wonderfulColshape) return;
    // Colshape is 'wonderfulColshape'
    ...
});

alt.on('entityEnterColshape', (colshape: Colshape, entity: Entity) => {
    if(!entity instanceof alt.Vehicle) return;
    // Entity is instance of alt.Vehicle

    if(colshape != 'wonderfulColshape) return;
    // Colshape is 'wonderfulColshape'
    ...
});

// OR

alt.on('entityEnterColshape', (colshape: Colshape, entity: Entity) => {
    if(!entity instanceof alt.Player || colshape != 'wonderfulColshape') return;

    // Entity is instance of alt.Player and colshape is 'wonderfulColshape'
    ...
});

alt.on('entityEnterColshape', (colshape: Colshape, entity: Entity) => {
    if(!entity instanceof alt.Vehicle || colshape != 'wonderfulColshape') return;

    // Entity is instance of alt.Vehicle and colshape is 'wonderfulColshape'
    ...
});
```

**Comment by AlteredCarbonate** [comment]
Implemented in #1280

**Comment by Lhoerion** [comment]
Added/~~Fixed~~/~~Changed~~ in build #[1250](https://discord.com/channels/371265202378899476/528297438348574741/735551360082116650)
